framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -259,31 +243,57 @@ int things_wifi_changed_call_func(int state, char *ap_name, char *ip_addr)
 
 static void *__attribute__((optimize("O0"))) t_things_wifi_join_loop(void *args)
 {
+	bool* is_wifi_retry_connect = (bool*) args;
+
+	if ((*is_wifi_retry_connect) == true) {
+		THINGS_LOG_E(TAG, "Retry to connect to home ap.");
+		things_network_connect_home_ap();
+	} else {
 #ifdef CONFIG_NETUTILS_NTPCLIENT
-	sync_time_from_ntp(10);
+		sync_time_from_ntp(10);
 #else
-	THINGS_LOG_V(TAG, "CONFIG_NETUTILS_NTPCLIENT is not set");
+		THINGS_LOG_V(TAG, "CONFIG_NETUTILS_NTPCLIENT is not set");
 #endif
+		wifi_manager_info_s wifi_info;
+		wifi_manager_get_info(&wifi_info);
 
-	wifi_manager_info_s wifi_info;
-	wifi_manager_get_info(&wifi_info);
-
-	things_wifi_changed_call_func(1, wifi_info.ssid, wifi_info.ip4_address);
+		things_wifi_changed_call_func(1, wifi_info.ssid, wifi_info.ip4_address);
```

Comment:
ip4.address seems that useless.

Reply:
in this case, the things was connected the wifi, so the ip4 address will be informed to easy-setup module.

Reply:
I mean in things_wifi_state_changed_cb, it just check log only.

Reply:
Yes, Let's check below logic 
There are some missing value here.
`int` things_wifi_state_changed_cb(int state, char *ap_name, char *ip_addr)`

Reply:
In this PR, we do focus on the reconnection issue better.
We need to separate issues. 
So I suggest to write an other P. R what function's parameters will be improved the side of useful.